deafrica_waterbodies/plugins/ocean_filtering_using_goas.py

```diff
@@ -0,0 +1,92 @@
+import os
+
+import geopandas as gpd
+import numpy as np
+import skimage.morphology
+import xarray as xr
+from deafrica_tools.spatial import xr_rasterize
+
+# File extensions to recognise as Parquet files.
+PARQUET_EXTENSIONS = {".pq", ".parquet"}
+
+buffer_pixels = 500 / 30
+
+
+def transform_hydrosheds_land_mask(hydrosheds_land_mask: xr.DataArray) -> xr.DataArray:
+    """
+    Function to transform the HydroSHEDs Land Mask into a boolean mask where
+    0/False are ocean pixels and 1/True are land pixels.
+    """
+    # Indicator values: 1 = land, 2 = ocean sink, 3 = inland sink, 255 is no data.
+    boolean_mask = (hydrosheds_land_mask != 255) & (hydrosheds_land_mask != 2)
+
+    return boolean_mask
+
+
+def erode_land_sea_mask(boolean_land_sea_mask: xr.DataArray, buffer_pixels: float) -> xr.DataArray:
+    """
+    Shrink the land in the land/sea mask.
+
+    Parameters
+    ----------
+    boolean_land_sea_mask : xr.DataArray
+        Boolean mask where 0/False are ocean pixels and 1/True are land pixels.
+    buffer_pixels : float
+        Number of pixels to erode the land by.
+
+    Returns
+    -------
+    xr.DataArray
+        Eroded land sea mask where 0/False are ocean pixels and 1/True are land pixels.
+    """
+    eroded_boolean_land_sea_mask = xr.apply_ufunc(
+        skimage.morphology.binary_erosion,
+        boolean_land_sea_mask,
+        skimage.morphology.disk(buffer_pixels),
+    )
+    return eroded_boolean_land_sea_mask
+
+
+def load_land_sea_mask(
+    land_sea_mask_fp: str,
+    wofs_alltime_summary_ds: xr.DataArray,
+) -> xr.DataArray:
+    """
+    Load the Marine Regions Global Oceans and Seas v01 from the file path
+    provided. Rasterize the vector data to match the loaded datacube WOfS
+    All Time Summary data and transform the raster into
+    a boolean mask where 0/False are ocean pixels and 1/True are land pixels.
+
+    Parameters
+    ----------
+    land_sea_mask_fp : str
+        File path to the Marine Regions Global Oceans and Seas v01 vector data.
+    wofs_alltime_summary_ds : xr.DataArray
+        Loaded datacube WOfS All Time Summary data to match to.
+
+    Returns
+    -------
+    xr.DataArray
+        A boolean land and sea mask from the Marine Regions Global Oceans and Seas v01 data.
+    """
+
+    _, file_extension = os.path.splitext(land_sea_mask_fp)
+    if file_extension not in PARQUET_EXTENSIONS:
+        land_sea_mask_gdf = gpd.read_file(land_sea_mask_fp).to_crs(
+            wofs_alltime_summary_ds.geobox.crs
+        )
+    else:
+        land_sea_mask_gdf = gpd.read_parquet(land_sea_mask_fp).to_crs(
+            wofs_alltime_summary_ds.geobox.crs
+        )
+
+    land_sea_mask_ds = xr_rasterize(land_sea_mask_gdf, wofs_alltime_summary_ds)
+    boolean_land_sea_mask = np.logical_not(land_sea_mask_ds)
+
+    # Erode the land in the land sea mask
+    eroded_boolean_land_sea_mask = erode_land_sea_mask(boolean_land_sea_mask, buffer_pixels)
+
+    return eroded_boolean_land_sea_mask
```

Comment:
May need to return the resolution of the eroded mask to use as part of the calculation of the number of pixels to erode.